plugins/vppv2/aclplugin/vppcalls/dump_vppcalls.go

```diff
@@ -0,0 +1,695 @@
+	}
+	return acls, nil
+}
+
+// DumpInterfaceMACIPAcls implements ACL handler.
+func (h *ACLVppHandler) DumpInterfaceMACIPAcls(swIndex uint32) (acls []*acl.Acl, err error) {
+	resMacIP, err := h.DumpInterfaceMACIPACLs(swIndex)
+	if err != nil {
+		return nil, err
+	}
+
+	if resMacIP.SwIfIndex != swIndex {
+		return nil, fmt.Errorf("returned interface index %d does not match request", resMacIP.SwIfIndex)
+	}
+
+	for aidx := range resMacIP.Acls {
+		macipACL, err := h.getMACIPACLDetails(uint32(aidx))
+		if err != nil {
+			return nil, err
+		}
+		acls = append(acls, macipACL)
+	}
+	return acls, nil
+}
+
+// DumpInterfaceIPACLs implements ACL handler.
+func (h *ACLVppHandler) DumpInterfaceIPACLs(swIndex uint32) (*acl_api.ACLInterfaceListDetails, error) {
+	req := &acl_api.ACLInterfaceListDump{
+		SwIfIndex: swIndex,
+	}
+	reply := &acl_api.ACLInterfaceListDetails{}
+
+	if err := h.dumpChannel.SendRequest(req).ReceiveReply(reply); err != nil {
+		return nil, err
+	}
+
+	return reply, nil
+}
+
+// DumpInterfaceMACIPACLs implements ACL handler.
+func (h *ACLVppHandler) DumpInterfaceMACIPACLs(swIndex uint32) (*acl_api.MacipACLInterfaceListDetails, error) {
+	req := &acl_api.MacipACLInterfaceListDump{
+		SwIfIndex: swIndex,
+	}
+	reply := &acl_api.MacipACLInterfaceListDetails{}
+
+	if err := h.dumpChannel.SendRequest(req).ReceiveReply(reply); err != nil {
+		return nil, err
+	}
+
+	return reply, nil
+}
+
+// DumpInterfaces implements ACL handler.
+func (h *ACLVppHandler) DumpInterfaces() ([]*acl_api.ACLInterfaceListDetails, []*acl_api.MacipACLInterfaceListDetails, error) {
+	msgIPACL := &acl_api.ACLInterfaceListDump{
+		SwIfIndex: 0xffffffff, // dump all
+	}
+
+	reqIPACL := h.dumpChannel.SendMultiRequest(msgIPACL)
+
+	var IPaclInterfaces []*acl_api.ACLInterfaceListDetails
+	for {
+		reply := &acl_api.ACLInterfaceListDetails{}
+		stop, err := reqIPACL.ReceiveReply(reply)
+		if stop {
+			break
+		}
+		if err != nil {
+			logrus.DefaultLogger().Error(err)
+			return nil, nil, err
+		}
+		IPaclInterfaces = append(IPaclInterfaces, reply)
+	}
+
+	msgMACIPACL := &acl_api.ACLInterfaceListDump{
+		SwIfIndex: 0xffffffff, // dump all
+	}
+
+	reqMACIPACL := h.dumpChannel.SendMultiRequest(msgMACIPACL)
+
+	var MACIPaclInterfaces []*acl_api.MacipACLInterfaceListDetails
+	for {
+		reply := &acl_api.MacipACLInterfaceListDetails{}
+		stop, err := reqMACIPACL.ReceiveReply(reply)
+		if stop {
+			break
+		}
+		if err != nil {
+			logrus.DefaultLogger().Error(err)
+			return nil, nil, err
+		}
+		MACIPaclInterfaces = append(MACIPaclInterfaces, reply)
+	}
+
+	return IPaclInterfaces, MACIPaclInterfaces, nil
+}
+
+func (h *ACLVppHandler) getIPRuleDetails(rule acl_api.ACLRule) (*acl.Acl_Rule, error) {
+	// Resolve rule actions
+	aclAction, err := h.resolveRuleAction(rule.IsPermit)
+	if err != nil {
+		return nil, err
+	}
+
+	// Resolve rule matches
+	match := &acl.Acl_Rule_Match{
+		IpRule: h.getIPRuleMatches(rule),
+	}
+
+	return &acl.Acl_Rule{
+		AclAction: aclAction,
```

Comment:
hmm, it seems that rule Name isn't actually configured/tagged and cannot be dumped (resync would consider the configured ACL different from what was dumped, thus calling Modify). I think the rule names should be removed from the model as they play no role (ACL is a list of rules, not a map anyway, i.e. the index in the list is ID if anything)

EDIT: I see you have already removed ruleName so we are on the same page :)